src/lib/libdylink.js

```diff
@@ -959,11 +980,46 @@ var LibraryDylink = {
      * @param {number=} handle
      * @param {Object=} localScope
      */`,
-  $loadDynamicLibrary: function(libName, flags = {global: true, nodelete: true}, localScope, handle) {
+  $loadDynamicLibrary: function(libName, flags = {global: true, nodelete: true}, localScope, rpath = {parentLibPath: '', paths: []}, handle) {
 #if DYLINK_DEBUG
     dbg(`loadDynamicLibrary: ${libName} handle: ${handle}`);
     dbg(`existing: ${Object.keys(LDSO.loadedLibsByName)}`);
 #endif
+
+#if FILESYSTEM && SUPPORT_RPATH
+      var foundFile = false;
```

Comment:
Indentation looks wrong here.

Also, shouldn't the existing checks for `LDSO.loadedLibsByName[libName];` and / or `handle` come first?